src/components/CourseWorkSelector.jsx

```diff
@@ -0,0 +1,77 @@
+import classnames from 'classnames';
+import React from 'react';
+import PropTypes from 'prop-types';
+import map from 'lodash/map';
+import partial from 'lodash/partial';
+
+
+function CourseWorkSelector({
+  courses,
+  isOpen,
+  selectedCourse,
+  onCreateAssignment,
+  onCloseCourseWorkSelector,
+  onSelectCourse,
+}) {
+  if (!isOpen) {
+    return null;
+  }
+  const courseOptions = map(courses, course => (
+    <option key={course.id} value={course.id}>{course.name}</option>
+  ));
```

Comment:
Per discussion, probably the easiest thing is to generate an extra first `<option>` that is just blank / empty string value, so that the Redux state and the form state are always in sync.